compiler-plugin/src/main/kotlin/it/unibo/collektive/frontend/checkers/CheckersUtility.kt

```diff
@@ -61,4 +63,87 @@ object CheckersUtility {
      */
     fun CheckerContext.isInsideAggregateFunction(): Boolean =
         containingElements.any { (it as? FirSimpleFunction)?.receiverParameter?.isAggregate(session) == true }
+
+    /**
+     * Returns wrapping [elements][FirElement] until it finds the element that satisfies the predicate (which is
+     * excluded from the result). The context's element is excluded.
+     *
+     * If [excludeDotCall] is set to *true*, elements that represent a dot call on the context's element are excluded
+     * as well. For example, in this code:
+     * ```kotlin
+     * for (...) {
+     *    for (...) {
+     *       <CONTEXT_ELEMENT>.map(...)
+     *    }
+     * }
+     * ```
+     * With [excludeDotCall] set to *false*, this method would return `List(FirFunctionCall("map"),
+     * FirWhileLoop("for"), FirWhileLoop("for"))`, instead with *true* the first `map` would be excluded.
+     */
+    fun CheckerContext.wrappingElementsUntil(
+        excludeDotCall: Boolean = true,
+        predicate: (FirElement) -> Boolean,
+    ): List<FirElement>? {
+        val calleeName = (containingElements.last() as? FirFunctionCall)?.functionName()
+        return containingElements.takeIf { it.any(predicate) }
+            ?.let { firElements ->
+                if (excludeDotCall) {
+                    firElements.filterNot {
+                        calleeName == it.receiverName()
+                    }
+                } else {
+                    firElements
+                }
+            }
+            ?.dropLast(1)
+            ?.takeLastWhile { !predicate(it) }
+    }
+
+    /**
+     * Gets the receiver's name of this element. For example, in this code:
+     * ```kotlin
+     * exampleFunction().map { ... }
+     * ```
+     * If we call this method on the `map` element, it returns `exampleFunction`.
+     * If, instead, we call this method on something that is not a function call or that has no receiver, it returns
+     * `null`.
+     */
+    fun FirElement.receiverName(): String? =
+        ((this as? FirFunctionCall)?.explicitReceiver?.unwrapExpression() as? FirFunctionCall)?.functionName()
+
+    /**
+     * Returns the receiver [List] only if it doesn't contain any function declaration, or `null` otherwise.
```

Comment:
On the return type, I suggest avoiding the `[List]` link since do not add any valuable insight.